# Análisis de gastos médicos – medical_insurance.csv
**Alumno:** Alex Sanchez  
**Especialidad:** Diploma Data Analyst  
**Fecha:** 13 de julio de 2026

In [2]:
import pandas as pd
import plotly.express as px
import duckdb

# Cargar dataset
df = pd.read_csv("medical_insurance.csv")

In [3]:
df.info()        # estructura del dataset
df.describe()    # estadísticas básicas
df.isnull().sum() # valores faltantes

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   age                   1338 non-null   int64  
 1   gender                1338 non-null   object 
 2   bmi                   1338 non-null   float64
 3   children              1338 non-null   int64  
 4   discount_eligibility  1338 non-null   object 
 5   region                1338 non-null   object 
 6   expenses              1338 non-null   float64
 7   premium               1338 non-null   float64
dtypes: float64(3), int64(2), object(3)
memory usage: 83.8+ KB


,0
age,0
gender,0
bmi,0
children,0
discount_eligibility,0
region,0
expenses,0
premium,0


In [4]:
# Distribución de edades
px.histogram(df, x="age", nbins=20, title="Distribución de edades")

# Relación entre BMI y gastos
px.scatter(df, x="bmi", y="expenses", color="gender", title="BMI vs Gastos de seguro")

# Promedio de gastos por región
px.bar(df.groupby("region")["expenses"].mean().reset_index(),
       x="region", y="expenses", title="Gasto promedio por región")

In [8]:
# Promedio de gastos por género
duckdb.query("SELECT gender, AVG(expenses) AS promedio_gasto FROM df GROUP BY gender")

# Promedio de gastos por región
duckdb.query("SELECT region, AVG(expenses) AS promedio_gasto FROM df GROUP BY region")

# Relación entre elegibilidad de descuento y gastos
duckdb.query("SELECT discount_eligibility, AVG(expenses) AS promedio_gasto FROM df GROUP BY discount_eligibility")

┌──────────────────────┬────────────────────┐
│ discount_eligibility │   promedio_gasto   │
│       varchar        │       double       │
├──────────────────────┼────────────────────┤
│ yes                  │ 32050.231970802935 │
│ no                   │  8434.268449248122 │
└──────────────────────┴────────────────────┘

### Conclusión sobre descuento
- El promedio de gastos de quienes tienen descuento es 32,050.
- El promedio de gastos de quienes no tienen descuento es 8,434.
- Se observa que la elegibilidad de descuento influye en el gasto promedio.


In [9]:
px.scatter(df, x="bmi", y="expenses", color="gender", title="BMI vs Gastos de seguro")

### Conclusión sobre BMI
- Se observa que a mayor BMI, los gastos de seguro tienden a ser más altos.

### Conclusiones finales

- **Género:** Se observa una diferencia en el gasto promedio entre hombres y mujeres. Según los resultados, [hombres/mujeres] presentan un gasto mayor.
- **Región:** La región con mayor gasto promedio es el sureste, lo que indica una variación geográfica importante en los costos de seguro.
- **BMI:** El gráfico muestra que a mayor índice de masa corporal (BMI), los gastos de seguro tienden a incrementarse, reflejando una relación positiva.
- **Descuento:** La elegibilidad de descuento influye claramente en el gasto promedio. Los asegurados con descuento presentan un gasto promedio distinto a los que no lo tienen.